src/UserRepository/PdoDatabase.php

```diff
@@ -173,6 +203,13 @@ protected function getUserDetails(string $identity) : array
         if (! $stmt->execute()) {
             return [];
         }
-        return $stmt->fetch(PDO::FETCH_ASSOC);
+
+        $userDetails = $stmt->fetch(PDO::FETCH_ASSOC);
+
+        if (! $userDetails) {
```

Comment:
I think we should put in an assertion here (should be `Assert::isMap()`), as we don't expect a `false` to be returned (failure case)